src/Type/Php/ParseUrlFunctionDynamicReturnTypeExtension.php

```diff
@@ -105,19 +123,43 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 			return $this->componentTypesPairedConstantsForLowercaseString[$componentType->getValue()] ?? new ConstantBooleanType(false);
 		}
 
+		if ($urlType->isUppercaseString()->yes()) {
+			return $this->componentTypesPairedConstantsForUppercaseString[$componentType->getValue()] ?? new ConstantBooleanType(false);
+		}
+
 		return $this->componentTypesPairedConstants[$componentType->getValue()] ?? new ConstantBooleanType(false);
 	}
 
-	private function createAllComponentsReturnType(bool $urlIsLowercase): Type
+	private function createAllComponentsReturnType(bool $urlIsLowercase, bool $urlIsUppercase): Type
 	{
+		if ($urlIsLowercase && $urlIsUppercase) {
+			if (
+				$this->allComponentsTogetherTypeForLowercaseString === null
+				&& $this->allComponentsTogetherTypeForLowercaseString === null
```

Comment:
You'll need a `allComponentsTogetherTypeForLowercaseAndUppercaseString` i'm afraid.

Cause you shouldn't rely on allComponentsTogetherTypeForLowercaseString or allComponentsTogetherTypeForUppercaseString.

But maybe it's overcomplicated to handle both lowercase-string and uppercase-string here cause an url with only non-letter caracter maybe doesn't exists ?

WDYT @pmjones ? Maybe we should drop the case lowercase-string&uppercase-string in ParseUrl function.

Reply:
> Maybe we should drop the case lowercase-string&uppercase-string in ParseUrl function.

I am *absolutely* good with that -- I started out by trying to have uppercase available wherever lowercase was, but ParseUrl has become a bit much. I'll put it back the way it was before uppercase.

Reply:
Removed in 356e0f810